WeightProducer/python/getWeightProducer_cff.py

```diff
@@ -812,7 +812,7 @@ def getWeightProducer(fileName,fastsim=False, pmssm=False):
     for sample in samples:
         if sample.name in fileName and sample.production in fileName:
             weightProducer.Method     = cms.string(sample.Method)
-            weightProducer.XS         = cms.double(sample.XS)
+            weightProducer.XS         = cms.double(sample.XS*sample.kFactor)
```

Comment:
have you confirmed this gives the correct values in all cases (e.g. XS_13TeV * kFactor_2016, etc.)?